static/src/javascripts/projects/commercial/modules/hosted/onward.js

```diff
@@ -1,54 +1,61 @@
 // @flow
 import config from 'lib/config';
-import mediator from 'lib/mediator';
 import fetchJson from 'lib/fetch-json';
 import fastdom from 'lib/fastdom-promise';
-import HostedCarousel from 'commercial/modules/hosted/onward-journey-carousel';
+import { initHostedCarousel } from './onward-journey-carousel';
 
-const loadOnwardComponent = (
+const getPlaceholdersFromDom = (dom = document): Array<any> =>
+    [].slice.call(dom.getElementsByClassName('js-onward-placeholder'));
+
+const generateUrlFromConfig = (
+    c: {
+        page: {
+            ajaxUrl: string,
+            pageId: string,
+            contentType: string,
+        },
+    } = config
+): string => `${c.page.ajaxUrl}/${c.page.pageId}/${c.page.contentType.toLowerCase()}/onward.json`;
+
+const insertHTMLfromPlaceholders = (
+    json: { html: string },
+    placeholders: Array<any>
+): void => {
+    placeholders[0].insertAdjacentHTML('beforeend', json.html);
+};
+
+export const loadOnwardComponent = (
     start: () => void,
-    stop: () => void
+    stop: () => void,
+    getPlaceholders: () => Array<any> = getPlaceholdersFromDom,
+    getUrl: () => string = generateUrlFromConfig,
+    getJson: (url: string, options: RequestOptions) => Promise<any> = fetchJson,
+    insertHtmlFn: (
+        json: { html: string },
+        placeholders: Array<any>
+    ) => void = insertHTMLfromPlaceholders,
```

Comment:
This is a really difficult one. So much imperative code, so much external state, really tough to test! I think you've made good progress though. 

I think it's easier to discuss if go through these parameters one by one:

- `start` and `stop`: no change, all cool
- `getPlaceholders`: agree with the extraction into a separate function, but I don't think it should be injectable for the sake of making the default easier to test. You could test this by side effects, by building a fake DOM with the appropriate elements and spying on `insertHtmlFn`, to see what it was called with
- `getUrl`: again, agree with the extraction but not with the injection. You can test this by mocking out a fake `config`, and then spying on `fetchJson` to see what it was called with
- `getJson`: we can avoid this injection entirely, mock out `lib/fetch-json` to ensure it gets called with the correct params
- `insertHtmlFn`: I agree that this can and probably should be injected
- `initHostedComponent`: Since we're importing this, we can mock it out in the tests and ensure it gets called

These changes will negate the need for exporting or injecting functions for the purpose of testing.  

Loving the removal of `mediator`! Again makes this module so much clearer.